### PR TITLE
Add latest news and what we do section to new org pages

### PR DESCRIPTION
**Pull request by andysellick**
This PR adds the 'latest news', 'what we do' and 'our policies' sections to the new organisation pages.

screen shot 2018-06-14 at 15 13 57

The code is currently not returning data for the 'latest' section from the correct source, but this will be addressed in a later PR.

URLs to test with:
- [org page with all three sections](https://govuk-collections-pr-690.herokuapp.com/government/organisations/office-of-the-secretary-of-state-for-wales)
- [org page with just the first two](https://govuk-collections-pr-690.herokuapp.com/government/organisations/attorney-generals-office)

Trello cards:
- https://trello.com/c/BdGlYIwl/175-add-latest-news-and-communication-section-to-organisation-page
- https://trello.com/c/vlrxzeFp/174-add-what-we-do-section-to-organisation-page
- https://trello.com/c/Qw7Y6CKE/177-add-our-policies-list-to-what-we-do-section

### CODE REVIEW

File: app/presenters/organisations/show_presenter.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    def initialize(organisation)
+      @org = organisation
+    end
+
+    def prefixed_title
+      prefix = needs_definite_article?(org.title) ? "the " : ""
+      (prefix + org.title)
+    end
+
+    def latest
```
Comment (vanitabarrett) [inline app/presenters/organisations/show_presenter.rb]: Would it make sense to have a 'News' or 'Documents' presenter, which this could then live in along with the featured news stuff. As we're splitting things into different presenters according to use, I'm not sure it makes sense for this to live here.

Comment (vanitabarrett) [reply app/presenters/organisations/show_presenter.rb]: If you do choose to keep it here though, it should have a name like `latest_documents` so it's clear what this actually does.

File: app/presenters/organisations/show_presenter.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      stories = []
```
Comment (vanitabarrett) [inline app/presenters/organisations/show_presenter.rb]: Not sure this is a good variable name when the section displays the content it's meant to (i.e: not just news stories). Maybe `documents`. Same for here: https://github.com/alphagov/collections/pull/690/files#diff-ce57b95cdb3de98ec1f5711ba7dc9f24R17

On a related note, might be worth updating the Trello and PR to reflect that this section is not related to supergroups any more.

File: app/presenters/organisations/show_presenter.rb [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+      org.ordered_featured_documents.each do |story|
```
Comment (vanitabarrett) [inline app/presenters/organisations/show_presenter.rb]: I think this could be done more simply using `.map`. We have an example of formatting data for use with the document_list component in another place in Collections: https://github.com/alphagov/collections/blob/master/app/presenters/supergroups/supergroup.rb#L67

Would be something like:

```
      stories = org.ordered_featured_documents.each.map do |story|
        data = {
          link: {
            text: story["title"],
            path: story["href"]
          },
          metadata: {
            document_type: story["document_type"]
            public_updated_at: (Date.parse(story["public_updated_at"]) if story["public_updated_at"]),
          }
        }

        data
      end
```

File: app/views/organisations/_show_organisation.html.erb [hunk 4]
```diff
@@ -49,16 +49,34 @@
   <% end %>
 </div>
 
+<% if @organisation.ordered_featured_documents.length %>
```
Comment (vanitabarrett) [inline app/views/organisations/_show_organisation.html.erb]: Should this check `latest` rather than ordered_featured_documents? We'll definitely need to make that change when we get the correct document data to show.

Comment (vanitabarrett) [reply app/views/organisations/_show_organisation.html.erb]: Also I'm not sure this works. It would need to be something like:

`if @organisation.ordered_featured_documents.length > 0`
or
`if @organisation.ordered_featured_documents.any?`

At the moment, even if length is 0 the section will still display.

File: app/views/organisations/_show_organisation.html.erb [hunk 5]
```diff
@@ -49,16 +49,34 @@
...
+  <%= render "govuk_publishing_components/components/heading", {
```
Comment (vanitabarrett) [inline app/views/organisations/_show_organisation.html.erb]: This will also need to go in a grid column-two-thirds.

File: app/views/organisations/_show_organisation.html.erb (hunk 4 above)
Comment (andysellick) [reply app/views/organisations/_show_organisation.html.erb]: Fixed.

File: app/views/organisations/_show_organisation.html.erb (hunk 5 above)
Comment (andysellick) [reply app/views/organisations/_show_organisation.html.erb]: Fixed.

File: app/presenters/organisations/what_we_do_presenter.rb [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+      end
+
+      {
+        stacked: true,
+        brand: org.brand,
+        links: links
+      }
+    end
+
+    def what_we_do_title
```
Comment (vanitabarrett) [inline app/presenters/organisations/what_we_do_presenter.rb]: Think this can just be called 'title', as when we reference it in  the view it's already prefixed with: `@what_we_do`

File: test/integration/organisation_test.rb [hunk 7]
```diff
@@ -178,10 +221,32 @@ class OrganisationHeaderTest < ActionDispatch::IntegrationTest
       }
     }
 
+    @content_item_blank = {
```
Comment (vanitabarrett) [inline test/integration/organisation_test.rb]: Not sure this belongs in this commit, especially as other tests are added separately.

File: app/presenters/organisations/show_presenter.rb [hunk 8]
```diff
@@ -0,0 +1,59 @@
+module Organisations
+  class ShowPresenter
```
Comment (vanitabarrett) [inline app/presenters/organisations/show_presenter.rb]: There is a lot of logic in this presenter, I think it's worth writing some tests for.

File: app/views/organisations/_show_organisation.html.erb [hunk 9]
```diff
@@ -79,12 +79,16 @@
   </div>
 
   <div class="grid-row">
-    <div class="column-two-thirds">
+    <div class="column-two-thirds brand--<%= @organisation.brand %>">
```
[the remark this answers is not in the document]
Comment (andysellick) [reply app/views/organisations/_show_organisation.html.erb]: It won't do it for some reason.

File: app/views/organisations/_show_organisation.html.erb [hunk 10]
```diff
@@ -46,7 +46,7 @@
     <div class="grid-row">
       <div class="column-two-thirds">
         <%= render "govuk_publishing_components/components/heading", {
-          text: "Latest from #{@show.prefixed_title}",
+          text: t('organisations.latest_from') + @show.prefixed_title,
```
[the remark this answers is not in the document]
Comment (andysellick) [reply app/views/organisations/_show_organisation.html.erb]: Ah, cool. Thanks.

File: test/integration/organisation_test.rb [hunk 11]
```diff
@@ -303,13 +312,35 @@ class OrganisationHeaderTest < ActionDispatch::IntegrationTest
...
   it "shows a large news item only on news organisations" do
     visit "/government/organisations/attorney-generals-office"
     assert page.has_css?(".gem-c-image-card.gem-c-image-card--large")
 
     visit "/government/organisations/charity-commission"
+    assert page.has_css?(".gem-c-image-card")
     refute page.has_css?(".gem-c-image-card.gem-c-image-card--large")
+
+    visit "/government/organisations/civil-service-resourcing"
+    refute page.has_css?(".gem-c-image-card")
```
[the remark this answers is not in the document]
Comment (andysellick) [reply test/integration/organisation_test.rb]: Good point.

File: app/presenters/organisations/what_we_do_presenter.rb (hunk 6 above)
Comment (andysellick) [reply app/presenters/organisations/what_we_do_presenter.rb]: Actually this is now duplicate code as it's been put into the much more generic show_presenter, so I'll remove it.

File: test/integration/organisation_test.rb (hunk 7 above)
Comment (andysellick) [reply test/integration/organisation_test.rb]: Which commit is this in?

File: test/integration/organisation_test.rb [hunk 12]
```diff
@@ -311,6 +311,7 @@ class OrganisationTest < ActionDispatch::IntegrationTest
 
   it "shows the no10 banner element only on no10's page" do
     visit "/government/organisations/prime-ministers-office-10-downing-street"
+    puts(page.html)
```
[the remark this answers is not in the document]
Comment (andysellick) [reply test/integration/organisation_test.rb]: D'oh

File: app/presenters/organisations/show_presenter.rb [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+    attr_reader :org
```
[the remark this answers is not in the document]
Comment (andysellick) [reply app/presenters/organisations/show_presenter.rb]: Yes, I think so. It gets called in prefixed_title - `prefix = needs_definite_article?(org.title) ? "the " : ""`

File: app/views/organisations/_show_organisation.html.erb [hunk 14]
```diff
@@ -23,32 +23,104 @@
...
-      <% news.each do |news_item| %>
-        <div class="column-one-third">
-          <%= render "govuk_publishing_components/components/image_card", news_item %>
-        </div>
+  <div class="grid-row">
+    <div class="column-two-thirds brand--<%= @organisation.brand %>">
+      <%= render "govuk_publishing_components/components/govspeak", {
+        } do %>
+          <p><%= @organisation.body.html_safe %></p>
+          <p>TODO: add the rest</p>
```
[the remark this answers is not in the document]
Comment (andysellick) [reply app/views/organisations/_show_organisation.html.erb]: Unsure, but I'll take it out and add a note to the card.

File: app/presenters/organisations/show_presenter.rb (hunk 13 above)
[the remark this answers is not in the document]
Comment (vanitabarrett) [reply app/presenters/organisations/show_presenter.rb]: You should be able to replace usages of org with @org

[the remark this answers is not in the document]
Comment (andysellick) [reply app/presenters/organisations/show_presenter.rb]: I know, but there's loads of them in the code and it makes it slightly cleaner.

File: test/presenters/organisations/show_presenter_test.rb [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+  it 'adds a prefix to a title when it should' do
+    content_item = ContentItem.new({ title: "Attorney General's Office" }.with_indifferent_access)
+    organisation = Organisation.new(content_item)
+    @show_presenter = Organisations::ShowPresenter.new(organisation)
+
+    expected = "the Attorney General's Office"
+    assert_equal expected, @show_presenter.prefixed_title
+  end
+
+  it 'does not add a prefix to a title when it should not' do
```
[the remark this answers is not in the document]
Comment (andysellick) [reply test/presenters/organisations/show_presenter_test.rb]: Added.

File: app/presenters/organisations/show_presenter.rb (hunk 13 above)
[the remark this answers is not in the document]
Comment (vanitabarrett) [reply app/presenters/organisations/show_presenter.rb]: It's a bit misleading though because it creates a getter method which could be called outside the class, which we don't need. Someone unfamiliar with the code may think this is an indication of it being needed elsewhere.

You could make it private, e.g:

```
private
    attr_reader :org

public
    #rest of presenter code goes here


```

File: app/presenters/organisations/show_presenter.rb (hunk 8 above)
Comment (andysellick) [reply app/presenters/organisations/show_presenter.rb]: Added. It'll need expanding and modifying over time, but it's started.
